### ISSUE TITLE
LdJump taking 5+ days to run?

### ISSUE THREAD

**Issue by virpatel524** [PhHermann/LDJump#2]
Hi, 

I'm running LDJump with the following parameters: 

LDJump("/gpfs/fs0/data/wraycompute/vdp5/empty-scratch/fasta_alignemnts/5iter_allvariants_filtered.rareremove.deploidcorrected.notrelevantremoved.nopirvirvariants.ann.vcf.LT635624.fasta", alpha = 0.05, segLength = 1000, pathLDhat = "/gpfs/fs0/data/wraycompute/vdp5/src/LDhat/", format = "fasta", refName = NULL, start = NULL,
       thth = 0.01, constant = F, status = T)

This is running over an entire chromosome of _P. vivax_. It's been running for way too long, so I think I must be doing something wrong. Do you have any ideas for how I can optimize for speed, or should I just wait things out? 

Thank you very much for your help!

**Comment by PhHermann** [comment]
Hi, 

thanks for your feedback. The runtime of **LDJump** depends on the number and length of the sequences. In the meantime I was working on an update of **LDJump** which is now able to run in parallel (Version 0.1.7). 

The additional parameter _cores_ reflects the number of cores on which **LDJump** should be run. You can add it to the command you provided and the runtime should decrease. Note that the progress of **LDJump** is no longer printed on screen for a parallel job, but it is written to the _LDJump_Status.txt_ file. 

Cheers, 
Philipp

**Comment by virpatel524** [comment]
Thanks so much Philipp! I'm excited to put the cores option to use. 

However, when I run this on my machine, I run into two issues. 

First, since I'm performing this on malaria data, the coverage is spotty. Many of the regions have no alleged SNPs. So I get lots of segments with no SNPs as indicated by the message: 

`There are 84 segments with less than 1 SNP. Do you want to continue and impute the estimated rate? We recommend to use larger segment lengths. Input here y/n:`

However, if I run this through an Rscript, the program automatically fails, since I am unable to enter Y or N beforehand. Would you be able to implement an optional argument to indicate whether or not to continue? 

Second, running things under 4 cores, I get the following issue repeatedly: 


`Error in opening sequence file

..now exiting to system...
head: cannot open `39outfile.txt' for reading: No such file or directory


Run-time error

Error in opening sequence file

..now exiting to system...


Run-time error

Error in opening sequence file

..now exiting to system...
head: cannot open `40outfile.txt' for reading: No such file or directory`

It runs fine when I use only one core, but this happens under a multicore model. Any idea on how to fix this issue?

**Comment by PhHermann** [comment]
Thanks again for your feedback. 

> However, if I run this through an Rscript, the program automatically fails, since I am unable to enter Y or N beforehand. Would you be able to implement an optional argument to indicate whether or not to continue?

I have included a parameter _accept_, which should solve the problem when running **LDJump** in an R-script. You can add to the command accept = T, and **LDJump** will continue with the calculation and provide you the estimated segments in the output anyways. This is implemented in version 0.1.8. 


> Second, running things under 4 cores, I get the following issue repeatedly:
>
> `Error in opening sequence file
>
> ..now exiting to system...
> head: cannot open 39outfile.txt for reading: No such file or directory

Although the package was running on our cluster in parallel without any problems this might be due to a problem with the directories. I have therefore made slight adaptions to commands changing the working directories. Please try your job again with the uploaded version 0.1.8. 

Cheers, 
Philipp

**Comment by stsmall** [comment]
@virpatel524, @PhHermann,
I am having the same issue with Run-time errors when using >1 core. I added the accept = T option as well, but it didnt seem to help. @virpatel524 did you find a solution or did @PhHermann suggestions solve your problem?

**Comment by PhHermann** [comment]
Hi @stsmall, 

thank you for your comments. I made adaptions to the code such that **LDJump** is working on my machine given that the full path to the *fasta*-file is provided. 

Do these changes fix your issue?

Cheers, 
Philipp

**Comment by stsmall** [comment]
Hi Philipp,
I updated to the master and the example now runs without error for multiple cores. 
However, I am now having an issue with my actual data in "vcf" format. I will open a new issue for that problem since this is closed.
thanks,
scott
